File: src/main/java/me/thevipershow/safechat/SafeChatPlugin.java
package me.thevipershow.safechat;

import lombok.Getter;
import me.thevipershow.safechat.api.config.ConfigManager;
import me.thevipershow.safechat.api.config.TomlConfigurations;
import me.thevipershow.safechat.api.sql.ISafeChatDB;
import org.bukkit.plugin.java.JavaPlugin;

@Getter
public final class SafeChatPlugin extends JavaPlugin {

    private ConfigManager configManager;
    private ISafeChatDB iSafeChatDB;

    private ISafeChatDB loadDatabase() {
        // TODO: finish coding UwU
        return null;
    }

    @Override
    public final void onEnable() {
        configManager = ConfigManager.getInstance(this);
        configManager.loadAllConfigs(TomlConfigurations.class);
    }

    @Override
    public final void onDisable() {

    }
}

File: src/main/kotlin/studio/thevipershow/safechat/SafeChatPlugin.kt
package studio.thevipershow.safechat

import org.bukkit.plugin.java.JavaPlugin
import studio.thevipershow.safechat.config.ConfigManager

/**
 * Main class.
 */
class SafeChatPlugin : JavaPlugin() {

    val configManager = ConfigManager(this)

    /**
     * Called when the plugin is loading.
     */
    override fun onLoad() {
        configManager.saveAllIfAbsent()
    }

    /**
     * Called when the plugin is enabling.
     */
    override fun onEnable() {

    }
}